include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+
+
+    /**
+     * For the solution vector identified by @p global_vector_name, compute
+     * the third_derivatives of the function at the quadrature points, and
+     * return a vector with the correct type deduced by the Extractor you passed
+     * as the @p variable argument.
+     *
+     * Before you can call this method, you need to call the
+     * extract_local_dof_values() method at least once, passing the same
+     * @p global_vector_name string, and the same type for the variable @p dummy.
+     *
+     * If you have not previously called the extract_local_dof_values() method,
+     * this function will throw an exception.
+     *
+     * For this function to work properly, the underlying FEValues,
+     * FEFaceValues, or FESubfaceValues object for which you called one of the
+     * reinit() functions, must have computed the information you are
+     * requesting. To do so, the update_3rd_derivatives flag must be an
+     * element of the list of UpdateFlags that you passed to the constructor of
+     * this object. See The interplay of UpdateFlags, Mapping, and FiniteElement
+     * in FEValues for more information.
+     */
+    template <typename Extractor, typename Number = double>
+    const std::vector<typename OutputType<dim, spacedim, Number, Extractor>::
+                        third_derivative_type> &
+    get_third_derivatives(const std::string &global_vector_name,
+                          const Extractor &  variable,
+                          const Number       dummy = Number(0));
+
+    /**
+     * For the solution vector identified by @p global_vector_name, compute
+     * the symmetric gradients of the function at the quadrature points, and
+     * return a vector with the correct type deduced by the Extractor you passed
+     * as the
+     * @p variable argument.
+     *
+     * Before you can call this method, you need to call the
+     * extract_local_dof_values() method at least once, passing the same
+     * @p global_vector_name string, and the same type for the variable @p dummy.
+     *
+     * If you have not previously called the extract_local_dof_values() method,
+     * this function will throw an exception.
+     *
+     * For this function to work properly, the underlying FEValues,
+     * FEFaceValues, or FESubfaceValues object for which you called one of the
+     * reinit() functions, must have computed the information you are
+     * requesting. To do so, the update_gradients flag must be an element of the
+     * list of UpdateFlags that you passed to the constructor of this object.
+     * See The interplay of UpdateFlags, Mapping, and FiniteElement in FEValues
+     * for more information.
+     */
+    template <typename Extractor, typename Number = double>
+    const std::vector<typename OutputType<dim, spacedim, Number, Extractor>::
+                        symmetric_gradient_type> &
+    get_symmetric_gradients(const std::string &global_vector_name,
```

Comment:
maybe move this after the `get_gradients` function?

Reply:
The rationale was: all functions that apply to scalars only. Then all the ones that apply to vectors. And so on.

Reply:
But if you prefer to have all first order operators first, then second order, etc., I'll move them up...

Reply:
I don't really care.